gnat2goto/driver/tree_walk.adb

```diff
@@ -694,31 +694,57 @@ package body Tree_Walk is
       Resolved_Underlying : constant Irep :=
         Follow_Symbol_Type (Underlying, Global_Symbol_Table);
       --  ??? why not get this from the entity
+
+      function Get_Bound_Expr (Bound_Node : Node_Id) return Bound_Type_Expr;
+      function Get_Bound_Expr (Bound_Node : Node_Id) return Bound_Type_Expr
+      is
+         Attribute_Not_Range_Bound : exception;
+      begin
+         case Get_Attribute_Id (Attribute_Name (Bound_Node)) is
+            when Attribute_First =>
+               return Bound_Type_Expr (Do_Array_First (Bound_Node));
+            when Attribute_Last  =>
+               return Bound_Type_Expr (Do_Array_Last (Bound_Node));
+               when others          => raise Attribute_Not_Range_Bound;
```

Comment:
Should probably be replaced with `assert`

Reply:
Made into a precondition and changed from `case` to `if`.